### PR TITLE
Add /_matrix/versions to spec

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: api/client-server/versions.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+swagger: '2.0'
+info:
+  title: "Matrix Client-Server Versions API"
+  version: "1.0.0"
+host: localhost:8008
+schemes:
+  - https
+  - http
+basePath: /_matrix
```
Comment (richvdh) [inline api/client-server/versions.yaml]: This either needs to be a generic api, returning information on both the C-S versions *and* the S-S versions (in which case you need to change the response format), or it needs mounting at `/_matrix/client`. I guess the latter makes sense.

File: api/client-server/versions.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+consumes:
+  - application/json
```
Comment (richvdh) [inline api/client-server/versions.yaml]: not really.

File: api/client-server/versions.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+paths:
+  "/versions":
+    get:
+      summary: Gets the versions of the specification supported by the server.
+      description: |-
+        Gets the versions of the specification supported by the server.
+
+        The version numbers reported are of form ``rx.y.z``.
+        
+        Changes to ``x`` and ``y`` should not be assumed to be compatible with any other version.
```
Comment (richvdh) [inline api/client-server/versions.yaml]: This is really stuff that ought to be elsewhere in the spec - probably in the [intro](https://matrix.org/docs/spec/r0.0.0/intro.html).  I'd prefer if you moved it, but would rather it were here than nowhere :)

File: api/client-server/versions.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+        For a fixed ``x`` and ``y``, any ``z`` should be assumed to be compatible with any lesser ``z``.
```
Comment (richvdh) [inline api/client-server/versions.yaml]: Also, you need to define 'compatible with' better. I think you mean 'clients based on spec x.y.z will be compatible with servers implementing x.y.Z, where Z > z'. Is the reverse also true (I don't think so)?

File: api/client-server/versions.yaml (hunk 1 above)
Comment (illicitonion) [inline api/client-server/versions.yaml]: Done

File: api/client-server/versions.yaml (hunk 2 above)
Comment (illicitonion) [inline api/client-server/versions.yaml]: Done

File: api/client-server/versions.yaml (hunk 3 above)
Comment (illicitonion) [inline api/client-server/versions.yaml]: Done

File: api/client-server/versions.yaml (hunk 4 above)
Comment (illicitonion) [inline api/client-server/versions.yaml]: Done

**Comment by richvdh** [comment]
lgtm other than the nits. fix them and ship it

**Pull request merged**